zerver/views/auth.py

```diff
@@ -312,6 +326,16 @@ def login_or_register_remote_user(request: HttpRequest, result: ExternalAuthResu
     redirect_to = get_safe_redirect_to(redirect_to, user_profile.realm.uri)
     return HttpResponseRedirect(redirect_to)
 
+def handle_action_key(request: HttpRequest, user_profile: UserProfile,
+                      key: str) -> HttpResponse:
+    try:
+        confirmation_object = get_confirmation_object(key)
+    except Confirmation.DoesNotExist:
+        return render(request, 'confirmation/link_does_not_exist.html')
+
+    if confirmation_object.type == Confirmation.EMAIL_CHANGE:
+        return confirm_email_change(request, key, user_profile)
```

Comment:
this isn't returning a response if the type is not `EMAIL_CHANGE`. Is mypy not catching this?

Reply:
mypy isn't catching it.(Not sure why but I guess all current code paths either give an invalid key or one that is of type EMAIL_CHANGE). Will add a `else` here and render `'confirmation/link_does_not_exist.html'`. Would it be fine or shall I make a new page for this case explaining what's the issue? (That should generally not happen I think but we should handle that case too)